### PR TITLE
preserve debug view layout

### LINKED ISSUE #5815: debug view state is not preserved if it is not visible between reloads
It's a regression. It was fixed in https://github.com/theia-ide/theia/pull/5665 I will prepare a PR which includes the fix commit from #5665

### PR DESCRIPTION
**Pull request by akosyakov**
#### What it does

fix #5815: preserve the debug view layout state between reloads

commits extracted from https://github.com/theia-ide/theia/pull/5665

#### How to test

- Change debug view layout
- Reload the browser
- Check that layout is restored
- Test with the debug view collapsed and not (i.e. another side view is opened instead)
- Pay attention that breakpoints are there

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to review in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by AlexTugarev** [comment]
Testing right now.

**Comment by AlexTugarev** [comment]
I tested as described above and it works nicely in general!

The only thing I was surprised about was the hight of the view at the bottom after expansion after reload ;-)

**Comment by AlexTugarev** [comment]
I think there is an issue with travis: `E: Unable to locate package oracle-java9-set-default`

File: packages/core/src/browser/view-container.ts [hunk 1]
```diff
@@ -623,25 +625,65 @@ export class ViewContainerPart extends BaseWidget {
...
+        }
+        super.onUpdateRequest(msg);
+    }
+
     protected onAfterAttach(msg: Message): void {
         if (!this.wrapped.isAttached) {
-            Widget.attach(this.wrapped, this.body);
+            MessageLoop.sendMessage(this.wrapped, Widget.Msg.BeforeAttach);
+            // tslint:disable-next-line:no-null-keyword
+            this.body.insertBefore(this.wrapped.node, null);
```
Comment (AlexTugarev) [inline packages/core/src/browser/view-container.ts]: what does this do? ;-)

Comment (akosyakov) [reply packages/core/src/browser/view-container.ts]: do you mean `null` is not necessary? otherwise it appends a widget to the part body

Comment (AlexTugarev) [reply packages/core/src/browser/view-container.ts]: thanks!

**Comment by akosyakov** [comment]
> I think there is an issue with travis: E: Unable to locate package oracle-java9-set-default

Do you think it is temporarily or we should do something about it?

**Pull request merged**